control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -660,6 +666,57 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return result, nil
 }
 
+func (r *HostedControlPlaneReconciler) healthCheckKASExternalEndpoint(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	serviceStrategy := util.ServicePublishingStrategyByTypeForHCP(hcp, hyperv1.APIServer)
+	if serviceStrategy == nil {
+		return fmt.Errorf("APIServer service strategy not specified")
+	}
+
+	var hostName string
+	if serviceStrategy.Type == hyperv1.Route {
+		// svc = manifests.RouterPublicService(hcp.Namespace)
+		internalRoute := manifests.KubeAPIServerInternalRoute(hcp.Namespace)
+		if err := r.Get(ctx, client.ObjectKeyFromObject(internalRoute), internalRoute); err != nil {
+			return fmt.Errorf("failed to get kube apiserver route: %w", err)
+		}
+
+		if len(internalRoute.Status.Ingress) == 0 || internalRoute.Status.Ingress[0].RouterCanonicalHostname == "" {
+			return fmt.Errorf("APIServer load balancer is not provisioned")
+		}
+		hostName = internalRoute.Status.Ingress[0].RouterCanonicalHostname
+	} else {
+		var svc *corev1.Service
+		if serviceStrategy.Type == hyperv1.LoadBalancer && util.IsPrivateHCP(hcp) {
```

Comment:
If the hcp is publicAndPrivate I think we'd want to report on both load balancers' health